ci/repokitteh/modules/ownerscheck.star

```diff
@@ -0,0 +1,224 @@
+def _get_specs(config):
+  return _get_relevant_specs(config.get('paths', []), github.pr_list_files())
+
+def _reconcile(config, specs=None):
+  specs = specs or _get_specs(config)
+
+  if not specs:
+    return []
+
+  approvers = _get_global_approvers()
+
+  print("approvers: %s" % approvers)
+
+  results = []
+
+  for spec in specs:
+    approved = _is_approved(spec, approvers)
+
+    print("%s -> %s" % (spec, approved))
+
+    results.append((spec, approved))
+
+    if spec.owner[-1] == '!':
+      _update_status(spec.owner[:-1], spec.prefix, approved)
+
+      if hasattr(spec, 'label'):
+        if approved:
+          github.issue_unlabel(spec.label)
+        else:
+          github.issue_label(spec.label)
+    elif hasattr(spec, 'label'): # fyis
+      github.issue_label(spec.label)
+
+  return results
+
+
+def _comment(config, results, force=False):
+  lines = []
+
+  for spec, approved in results:
+    if approved:
+      continue
+
+    mention = spec.owner
+
+    if mention[0] != '@':
+      mention = '@' + mention
+
+    if mention[-1] == '!':
+      mention = mention[:-1]
+
+    prefix = spec.prefix
+    if prefix:
+      prefix = ' for changes made to `' + prefix + '`'
+
+    mode = spec.owner[-1] == '!' and 'approval' or 'fyi'
+
+    key = "ownerscheck/%s/%s" % (spec.owner, spec.prefix)
+
+    if (not force) and (store_get(key) == mode):
+      mode = 'skip'
+    else:
+      store_put(key, mode)
+
+    if mode == 'approval':
+      lines.append('CC %s: Your approval is needed%s.' % (mention, prefix))
+    elif mode == 'fyi':
+      lines.append('CC %s: FYI only%s.' % (mention, prefix))
+
+  if lines:
+    github.issue_create_comment('\n'.join(lines))
+
+
+def _reconcile_and_comment(config):
+  _comment(config, _reconcile(config))
+
+
+def _force_reconcile_and_comment(config):
+  _comment(config, _reconcile(config), force=True)
+
+
+def _pr(action, config):
+  if action in ['synchronize', 'opened']:
+    _reconcile_and_comment(config)
+
+
+def _pr_review(action, review_state, config):
+  if action != 'submitted' or not review_state:
+    return
+
+  _reconcile(config)
+
+
+# Partial approvals are done by commenting "/lgtm [label]".
+def _lgtm_by_comment(config, comment_id, command, sender, sha):
+  labels = command.args
+
+  if len(labels) != 1:
+    react(comment_id, 'please specify a single label can be specified')
+    return
+
+  label = labels[0]
+
+  specs = [s for s in _get_specs(config) if hasattr(s, 'label') and s.label == label]
+
+  if len(specs) == 0:
+    react(comment_id, 'no relevant owners for "%s"' % label)
+    return
+
+  for spec in specs:
+    _store_partial_approval(sender, spec.files)
+
+  react(comment_id, None)
+
+  _reconcile(config, specs)
+
+
+handlers.pull_request(func=_pr)
+handlers.pull_request_review(func=_pr_review)
+
+handlers.command(name='checkowners', func=_reconcile)
+handlers.command(name='checkowners!', func=_force_reconcile_and_comment)
+handlers.command(name='lgtm', func=_lgtm_by_comment)
```

Comment:
Why is this file included? Maybe merge master?

Reply:
I think so.. not sure why it got added to the file, should I delete the "modules" folder?

Reply:
merge master would fix it. the file exists upstream, maybe you rebased?

Reply:
yea I did rebase.. haha thanks!